### PR TITLE
Cleanup deprecated methods from SPI

### PR DESCRIPTION
**Pull request by Sanne**
- https://hibernate.atlassian.net/browse/HSEARCH-2770

I verified each step foollowing along with the Infinispan project, so I know that these are safe now :)

### CODE REVIEW

File: engine/src/main/java/org/hibernate/search/spi/SearchFactoryBuilder.java [hunk 1]
```diff
@@ -1,40 +0,0 @@
...
-package org.hibernate.search.spi;
-
-import org.hibernate.search.cfg.spi.SearchConfiguration;
-import org.hibernate.search.engine.integration.impl.ExtendedSearchIntegrator;
-
-/**
- * @deprecated Use SearchIntegrationBuilder instead. This class will be removed!
- */
-@Deprecated
-public class SearchFactoryBuilder extends SearchIntegratorBuilder {
```
Comment (yrodiere) [inline engine/src/main/java/org/hibernate/search/spi/SearchFactoryBuilder.java]: I don't know if it was related, but maybe you could also remove the deprecated `SearchFactoryIntegrator`?

**Comment by Sanne** [comment]
pushed a new version

- fixed the javadoc
- removed `SearchFactoryIntegrator`

No idea why the javadoc didn't fail locally?

File: engine/src/main/java/org/hibernate/search/bridge/util/impl/ContextualExceptionBridgeHelper.java [hunk 2]
```diff
@@ -46,12 +45,6 @@
 	private final StringConversionContextImpl stringAdapter = new StringConversionContextImpl();
 
 	@Override
-	@Deprecated
-	public ConversionContext setClass(Class<?> clazz) {
-		return setConvertedTypeId( new PojoIndexedTypeIdentifier( clazz ) );
-	}
-
-	@Override
 	public ConversionContext setConvertedTypeId(IndexedTypeIdentifier type) {
```
Comment (Sanne) [inline engine/src/main/java/org/hibernate/search/bridge/util/impl/ContextualExceptionBridgeHelper.java]: it's a good point but I'll do that later:
I'll send some more PRs later with more deprecation changes and will include a proposal for this.

**Comment by Sanne** [comment]
ok, CI confirmed it's green: I'll merge it. Thanks for the review!

**Pull request merged**